fastlane/lib/fastlane/fast_file.rb

```diff
@@ -338,7 +338,15 @@ def import_from_git(url: nil, branch: 'HEAD', path: 'fastlane/Fastfile', version
             UI.user_error!("No tag found matching #{version.inspect}") if checkout_param.nil?
           end
 
-          Actions.sh("cd #{clone_folder.shellescape} && git checkout #{checkout_param.shellescape} #{checkout_path}")
+          if is_eligible_for_caching && version.nil?
+            # Update the repo if it's eligible for caching but the version isn't specified
+            UI.message("Fetching remote git branches and updating git repo...")
+            Helper.with_env_values('GIT_TERMINAL_PROMPT' => '0') do
+              Actions.sh("cd #{clone_folder.shellescape} && git fetch --all -q && git checkout #{checkout_param.shellescape} #{checkout_path} && git reset --hard && git merge")
```

Comment:
```suggestion
              Actions.sh("cd #{clone_folder.shellescape} && git fetch --all --quiet && git checkout #{checkout_param.shellescape} #{checkout_path} && git reset --hard && git merge")
```

I personally prefer to use the long versions of shell arguments, because they're clearer even for people that are new to the respective command.

Reply:
Totally agree with this. Will update it.